### ISSUE TITLE
关于在评估时候以及训练使用图片分辨率的问题

### ISSUE THREAD

**Issue by liuyuangeng7** [kaikai23/MaskGaussian#16]
作者你好！我是一位非常青睐于你的方法的读者，我在复现你的实验的时候发现某些指标和论文中报告的数值存在差距（如counter/flowers/bonsai/treehill等几个场景的lpips与论文中的数值不一致），我很期待你告知我你是怎么评估的，我有看到你在回复其他朋友的时候提到使用的是VGG主干网。另外一个问题就是你在训练是使用的是默认分辨率=-1，这意味着图像的最大边被调整为<= 1600像素，这是否说明你和标准3DGS所使用的数据集分辨率存在差距，在标准3DGS的论文中，作者的项目中提到对于MipNeRF360数据集默认标准为室内使用基于原图二倍下采样的分辨率即数据集中场景的image_2，而室外使用基于原图四倍下采样的分辨率即数据集中场景的image_4。你这种统一的最大边被调整为<= 1600像素，是否没有满足统一的评估条件？

**Comment by kaikai23** [comment]
你好呀，感谢对这个项目的欣赏！
项目里的评测部分的代码我都是直接搬的原始的3dgs仓库的，我只在上面加了我的mask相关的代码以及CUDA相关的更改。评测里面的lpips是使用什么网络，分辨率是多少，这些我都是直接照搬的，没有专门的改动哦。不过你提到的3DGS论文里默认使用image_2，这个可能略有不同。我在评测我的方法以及在3DGS仓库评测3DGS的时候，都是用的这样的命令 python train.py -s xxx -m xxx --eval，没有传resolution参数，所以resolution应该都是-1。在其他很多3dgs的论文里感觉大家也都是这样运行的, 所以应该是没有问题的。论文里的主表Table 1里的下面四个方法都是用这样的命令跑出来的，包括其他ablation所有的实验也都是。

你测出来的lpips是多少呀？以及你的显卡和pip list环境方便贴一下吗～

**Comment by liuyuangeng7** [comment]
作者你好，我所用的显卡是单卡的RTX 5090，另外我在复现你的代码到counter场景的时候LPIPS稍高，PSNR有所下降：
Setting up [LPIPS] perceptual loss: trunk [vgg], v[0.1], spatial [off]
/root/miniconda3/envs/envGsplat/lib/python3.10/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/root/miniconda3/envs/envGsplat/lib/python3.10/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Loading model from: /root/miniconda3/envs/envGsplat/lib/python3.10/site-packages/lpips/weights/v0.1/vgg.pth
 LPIPS 初始化模式: 标准 (net='vgg')
 计算中: counter ...
                                                                                                                                                                                                                                                                                                               
====================
Metric    counter
--------------------
PSNR        28.78   
SSIM        0.889   
LPIPS       0.293   
#GS (M)      0.35   
====================

而pip list是：
(envGsplat) root@autodl-container-6dda4bb49e-0df1cf43:~/autodl-tmp/MyProject/Mask-GS 2# pip list
Package                     Version
--------------------------- -----------
colorama                    0.4.6
diff_gaussian_rasterization 0.0.0
filelock                    3.20.0
fsspec                      2025.12.0
gsplat                      1.0.0
jaxtyping                   0.3.3
Jinja2                      3.1.6
joblib                      1.5.2
lpips                       0.1.4
markdown-it-py              4.0.0
MarkupSafe                  3.0.3
mdurl                       0.1.2
mpmath                      1.3.0
networkx                    3.4.2
ninja                       1.13.0
numpy                       2.2.6
nvidia-cublas-cu12          12.8.4.1
nvidia-cuda-cupti-cu12      12.8.90
nvidia-cuda-nvrtc-cu12      12.8.93
nvidia-cuda-runtime-cu12    12.8.90
nvidia-cudnn-cu12           9.10.2.21
nvidia-cufft-cu12           11.3.3.83
nvidia-cufile-cu12          1.13.1.3
nvidia-curand-cu12          10.3.9.90
nvidia-cusolver-cu12        11.7.3.90
nvidia-cusparse-cu12        12.5.8.93
nvidia-cusparselt-cu12      0.7.1
nvidia-nccl-cu12            2.27.5
nvidia-nvjitlink-cu12       12.8.93
nvidia-nvshmem-cu12         3.3.20
nvidia-nvtx-cu12            12.8.90
opencv-python               4.12.0.88
pandas                      2.3.3
pillow                      12.0.0
pip                         25.3
plyfile                     1.1
psutil                      7.1.3
Pygments                    2.19.2
python-dateutil             2.9.0.post0
pytz                        2025.2
rich                        14.2.0
scipy                       1.15.3
setuptools                  80.9.0
simple_knn                  0.0.0
six                         1.17.0
sympy                       1.14.0
torch                       2.9.1
torchvision                 0.24.1
tqdm                        4.67.1
triton                      3.5.1
typing_extensions           4.15.0
tzdata                      2025.2
wadler_lindig               0.1.7
wheel                       0.45.1

**Comment by kaikai23** [comment]
hi, 抱歉回复得有些慢。你这个counter出来的psnr/ssim/lpip是28.78， 0.889， 0.293，这个差距太大了肯定不对。我又重新git clone, 按照说明pip install然后跑了一下python run_all.py，得到的结果是这样 psnr/ssim/lpips = 29.00, 0.913, 0.188. 高斯的数量是329184.

<img width="1372" height="792" alt="Image" src="https://github.com/user-attachments/assets/a57dd65d-3b88-4b5f-9943-f44fbc92466c" />

我注意到你的环境里好像没有安装`mask_diff_gaussian_rasterization`这个包诶，你可以按照安装说明安装一下这个包吗？这篇论文的主要改动就是在这个包里，没安装这个包跑出来肯定是有问题的

**Comment by kaikai23** [comment]
我先关闭这个issue了，如果仍有问题请随时继续提问～
